### ISSUE TITLE
🐛 Formatting restores deleted text

### ISSUE THREAD

**Issue by MichaReiser** [rome/tools#2953]
### Environment information

```bash
* Code OSS (Open Source build of VS Code)
* Rome LSP (main)
* Arch x64
```

### What happened?

1. Enable "Format on Save" 
![image](https://user-images.githubusercontent.com/1203881/181495305-8fcea1ac-a26a-426e-8d3a-bad506b64720.png)
2. Type some text, hit save
3. Delete some text with backspace
4. Hit Save

The formatter sometime "reverts" the deletion changes and re-inserts the tokens. 

[Kooha-07-28-2022-13-30-42.webm](https://user-images.githubusercontent.com/1203881/181495224-812c3445-64f0-4012-865d-e9a089af25fd.webm)

### Expected result

Formatter formats the current code and not previous versions of it.

### Code of Conduct

- [X] I agree to follow Rome's Code of Conduct

**Comment by MichaReiser** [comment]
@leops any idea what's going on?

**Comment by ematipico** [comment]
Are you able to see the `Rome trace` channel output? We can check the order of the events sent by the LSP, maybe it will help us to triage the issue

**Comment by leops** [comment]
This might be another document synchronization issue like in #2932, but the root cause for this should have been fixed in `vscode-languageclient@8.0.2`. Did you also build the VSCode extension from the latest version ?

**Comment by MichaReiser** [comment]
That fixes it. Sorry for not having tried that first. However, I now get the following error on save if the file contains syntax errors. 

```
[Error - 15:29:33] Request textDocument/formatting failed.
  Message: the file could not be formatted since it has syntax errors and `format_with_errors` is disabled
  Code: -32603 
the file could not be formatted since it has syntax errors and `format_with_errors` is disabled
```

**Comment by IWANABETHATGUY** [comment]
> 

Do you enable the config `  "rome.formatter.formatWithSyntaxErrors": true,` ?

**Comment by ematipico** [comment]
Fixed in  #2956
